docs/troubleshooting.md

```diff
@@ -0,0 +1,249 @@
+```
+
+Great a `CSV` was created but it is in the failing state.
+
+## ClusterServiceVersion Troubleshooting
+
+Continuing from the scenario described in the [Subscription Troubleshooting section](#subscription-troubleshooting), your cluster now has a `CSV` in the failed phase. The `CSV` has been placed in the failed phase because OLM encountered an unrecoverable error when attempting to install the operator.
+
+Our first step should be to check the status of the `CSV`. The OLM operator will constantly update the Status with useful information regarding the state of the `CSV`. You can check the status of your `CSV` with the following command:
+
+```sh
+$ kubectl -n my-catalogsource-namespace get csv prometheusoperator.0.32.0 -o yaml | yq r - status
+certsLastUpdated: null
+certsRotateAt: null
+conditions:
+- lastTransitionTime: "2019-11-21T14:45:58Z"
+  lastUpdateTime: "2019-11-21T14:45:58Z"
+  message: csv in namespace with no operatorgroups
+  phase: Failed
+  reason: NoOperatorGroup
+lastTransitionTime: "2019-11-21T14:45:58Z"
+lastUpdateTime: "2019-11-21T14:46:40Z"
+message: csv in namespace with no operatorgroups
+phase: Failed
+reason: NoOperatorGroup
+```
+
+>Note: It is possible that the Status is missing, which suggests that the OLM operator is encountering an issue when processing the `CSV` in a very early stage. You should respond by reviewing the logs of the OLM operator.
+
+From the information within the `status.reason` and `status.message` fields, we now know that the `CSV` failed to install because it has been deployed in a namespace without an [OperatorGroup](https://operator-framework.github.io/olm-book/docs/operator-scoping).
+
+Let's create an `OperatorGroup` in the `my-catalogsource-namespace namespace` by applying the following yaml:
+```sh
+$ cat operatorgroup.yaml
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: my-operators
+  namespace: my-catalogsource-namespace
+```
+
+Now we should inspect the status of the `CSV` again:
+```
+kubectl -n my-catalogsource-namespace get csvs
+NAME                        DISPLAY               VERSION   REPLACES                    PHASE
+prometheusoperator.0.32.0   Prometheus Operator   0.32.0    prometheusoperator.0.27.0   Failed
+```
+
+The `CSV` is still in the failed state. Let's reinspect the `reason` and `message` within the status block of the `CSV` with the following commands:
+```sh
+$ kubectl -n my-catalogsource-namespace get csv prometheusoperator.0.32.0 -o yaml | yq r - status.reason
+reason: UnsupportedOperatorGroup
+
+$ kubectl -n my-catalogsource-namespace get csv prometheusoperator.0.32.0 -o yaml | yq r - status.message
+message: AllNamespaces InstallModeType not supported, cannot configure to watch all namespaces
+```
+
+From the output above we can derrive that the `CSV` is failing to install because it has been deployed in a namespace that includes an `OperatorGroup` that whose installMode is not supported by the `CSV`.
```

Comment:
FYI, I think we decided not to discuss APIs like `OperatorGroup` since these are user facing docs, and not dev facing.

Reply:
I believe an `OperatorGroup` is a customer facing API.

Reply:
Sorry, I meant to say we were trying to make sure that these docs are use case driven, and not API or feature driven. But I guess when debugging an installation you need to mention the APIs.